### PR TITLE
Upgrade to Babel 7

### PR DESCRIPTION
**Pull request by colin-aarts**
Upgrade to Babel 7

 * Upgrade all official Babel packages to version `^7`;
 * Upgrade packages that depend on babel so they work with babel 7
 * Remove manual construction and inclusion of babel's `external-helpers` in favour of `@babel/plugin-transform-runtime` which automagically transforms calls to helper functions into imports (per bundle);
 * Remove duplicate browserslist config from `.babelrc` as babel 7 reads from `.browserslistrc`;
 * Update `.gitignore` to include VS Code files.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,15 +42,17 @@
     "tabWidth": 2
   },
   "devDependencies": {
+    "@babel/core": "^7.0.0",
```
Comment (awolkers) [inline package.json]:
```suggestion
    "@babel/core": "^7.1.2",
```
Maybe update to latest versions?

**Comment by wolfpilot** [comment]
@colin-aarts Can you please rebase your work on `master` and push again? Seems like you're pushing an older Git tree, including a few already merged PRs. Thanks!

File: package.json (hunk 1 above)
Comment (colin-aarts) [reply package.json]: Good call. I used https://github.com/babel/babel-upgrade but that tool seems to have `7.0.0` hardcoded :) Will do.

**Comment by colin-aarts** [comment]
Babel packages updated to latest versions.

**Comment by colin-aarts** [comment]
@wolfpilot Nice catch, done!

**Pull request merged**